src/Common/HashTable/HashTable.h

```diff
@@ -652,6 +658,15 @@ class HashTable : private boost::noncopyable,
                 ///   [1]: https://github.com/google/sanitizers/issues/854#issuecomment-329661378
                 __msan_unpoison(it.ptr, sizeof(*it.ptr));
             }
+
+            /// Everything had been destroyed in the loop above, reset the flag
+            /// only, without calling destructor.
+            this->clearHasZeroFlag();
+        }
+        else
+        {
+            if (this->hasZero())
+                this->clearHasZero();
```

Comment:
If this is branch where Cell is not trivially destructible, can't we call `clearHasZero` unconditionally ?